### DISCUSSION TITLE
Lazy Load React Server Components

### DISCUSSION THREAD

**Discussion by laurent512** [vercel/next.js#58021] [App Router]
Hello, 

For a dashboard like application, I am displaying a list of items in a table. 
For each of these items, the user should be able to  click on the "view details" button, which will open a modal with extra details. 

Viewing items details requires database fetch, and I wanted to implement it using server components. 
=> "view details" is a rsc , which will query the db at render time, (can be seen as a server side fetch)  

Problem is that the list of items is relatively long, and I don't want to overload the database by loading the details of each item.
I would like to lazy load the item details if someone click on it. 

Problem I stumbled upon is that dynamic import does not work for server components. 
How to deal with that ? What is the proper pattern to conditionally render react server components ?

**Comment by laurent512** [comment]
any opinion about it? :)

**Comment by zipme** [comment]
I am in the same situation. You could probably use [intercepting-routes](https://nextjs.org/docs/app/building-your-application/routing/intercepting-routes) and load your data there. But I was hoping there could be an easier solution without involving changing the url
